agents-api/agents_api/routers/tasks/routers.py

```diff
@@ -0,0 +1,100 @@
+from typing import Annotated, Literal
+from uuid import uuid4
+from agents_api.models.execution.create_execution import create_execution_query
+from agents_api.models.execution.get_execution_status import get_execution_status_query
+from agents_api.models.execution.get_execution_transition import (
+    get_execution_transition_query,
+)
+from agents_api.models.task.create_task import create_task_query
+from agents_api.models.task.get_task import get_task_query
+from agents_api.models.task.list_tasks import list_tasks_query
+from fastapi import APIRouter, HTTPException, status, BackgroundTasks, Depends
+import pandas as pd
+from pydantic import BaseModel
+from pydantic import UUID4
+from starlette.status import HTTP_201_CREATED, HTTP_202_ACCEPTED
+
+from pycozo.client import QueryException
+from agents_api.autogen.openapi_model import (
+    CreateTask,
+    CreateExecution,
+    Task,
+    Execution,
+    ExecutionTransition,
+    ResourceCreatedResponse,
+    ResourceUpdatedResponse,
+)
+from agents_api.dependencies.developer_id import get_developer_id
+
+
+class TaskList(BaseModel):
+    items: list[Task]
+
+
+router = APIRouter()
+
+
+@router.get("/tasks", tags=["tasks"])
+async def list_sessions(
+    x_developer_id: Annotated[UUID4, Depends(get_developer_id)],
+) -> TaskList:
+    query_results = list_tasks_query(x_developer_id)
+    return TaskList(
+        items=[Task(**row.to_dict()) for _, row in query_results.iterrows()]
+    )
+
+
+@router.get("/tasks/{task_id}", tags=["tasks"])
+async def get_task(
+    task_id: UUID4, x_developer_id: Annotated[UUID4, Depends(get_developer_id)]
+) -> Task:
+    query_results = get_task_query(x_developer_id, task_id)
+    pass
+
+
+@router.post("/tasks", status_code=HTTP_201_CREATED, tags=["tasks"])
+async def create_task(
+    request: CreateTask,
+    x_developer_id: Annotated[UUID4, Depends(get_developer_id)],
+) -> ResourceCreatedResponse:
+    task_id = uuid4()
+    resp: pd.DataFrame = create_task_query(
+        task_id=task_id,
+        developer_id=x_developer_id,
+    )
+    return ResourceCreatedResponse(
+        id=resp["task_id"][0], created_at=resp["created_at"][0]
+    )
+
+
+@router.post("/tasks/{task_id}/execution", status_code=HTTP_201_CREATED, tags=["tasks"])
```

Comment:
convention is that every resource name must be plural so /tasks/{task_id}/**executions**

Reply:
done